ctest-next/src/ast/constant.rs

```diff
@@ -0,0 +1,41 @@
+/// Represents a constant variable defined in Rust.
+pub struct Const {
+    public: bool,
+    ident: syn::Ident,
+    ty: syn::Type,
+    expr: syn::Expr,
+}
+
+impl Const {
+    /// Creates a new constant.
+    pub fn new(public: bool, ident: syn::Ident, ty: syn::Type, expr: syn::Expr) -> Self {
```

Comment:
`new` doesn't need to be crate-public does it?

Reply:
Honestly I think you could probably delete `new` and all accessors except `.ident()` (which I assume will be needed for filters) for all ast types, and make the fields `pub(crate)`. Should be a bit easier to work around, especially when these functions don't do any logic yet.